In [4]:
import pandas as pd
import sqlite3

In [8]:
albums_query = \
"""
SELECT * 
FROM albums
"""

with sqlite3.connect('data/chinook.db') as chin_con: 
    albums_df = pd.read_sql(albums_query, chin_con)

albums_df.head()

AlbumId                                  Title  ArtistId
0        1  For Those About To Rock We Salute You         1
1        2                      Balls to the Wall         2
2        3                      Restless and Wild         2
3        4                      Let There Be Rock         1
4        5                               Big Ones         3

In [6]:
print(albums_df.shape)

(347, 3)


4. The number of records in the albums table is 347. 

In [12]:
albums_query = \
"""
SELECT DISTINCT Title
FROM albums
"""

with sqlite3.connect('data/chinook.db') as chin_con: 
    albums_df = pd.read_sql(albums_query, chin_con)

albums_df

Title
0                For Those About To Rock We Salute You
1                                    Balls to the Wall
2                                    Restless and Wild
3                                    Let There Be Rock
4                                             Big Ones
..                                                 ...
342                             Respighi:Pines of Rome
343  Schubert: The Late String Quartets & String Qu...
344                                Monteverdi: L'Orfeo
345                              Mozart: Chamber Music
346  Koyaanisqatsi (Soundtrack from the Motion Pict...

[347 rows x 1 columns]

5a) There are 347 albums in the database.

In [13]:
albums_query = \
"""
SELECT DISTINCT Artistid
FROM albums
"""

with sqlite3.connect('data/chinook.db') as chin_con: 
    albums_df = pd.read_sql(albums_query, chin_con)

albums_df

ArtistId
0           1
1           2
2           3
3           4
4           5
..        ...
199       271
200       272
201       273
202       274
203       275

[204 rows x 1 columns]

5b) There are 204 unique artists in the albums table.

In [18]:
media_types_query = \
"""
SELECT DISTINCT *
FROM media_types
"""

with sqlite3.connect('data/chinook.db') as chin_con: 
    media_types_df = pd.read_sql(media_types_query, chin_con)

media_types_df

MediaTypeId                         Name
0            1              MPEG audio file
1            2     Protected AAC audio file
2            3  Protected MPEG-4 video file
3            4     Purchased AAC audio file
4            5               AAC audio file

5c) There are 5 different types of media. 

In [20]:
media_types_query = \
"""
SELECT DISTINCT MediaTypeId 
FROM tracks
"""

with sqlite3.connect('data/chinook.db') as chin_con: 
    media_types_df = pd.read_sql(media_types_query, chin_con)

media_types_df

MediaTypeId
0            1
1            2
2            3
3            4
4            5

5d) All of the media types are represented in the tracks table

In [29]:
media_types_query = \
"""
SELECT MediaTypeId, COUNT(*) AS "number of tracks" 
FROM tracks
GROUP BY MediaTypeId

"""

with sqlite3.connect('data/chinook.db') as chin_con: 
    media_types_df = pd.read_sql(media_types_query, chin_con)

media_types_df

MediaTypeId  number of tracks
0            1              3034
1            2               237
2            3               214
3            4                 7
4            5                11

5e) Media type 1 has 3034 tracks, media type 2 has 237 tracks, media type 3 has 214 tracks, media type 4 has 7 tracks, and media type 5 has 11 tracks. 

In [39]:
customers_query = \
"""
SELECT City, COUNT(*) AS "number of customers"
FROM customers
GROUP BY City
ORDER BY "number of customers" DESC

"""

with sqlite3.connect('data/chinook.db') as chin_con: 
   customers_df = pd.read_sql(customers_query, chin_con)

customers_df

City  number of customers
0             São Paulo                    2
1                Prague                    2
2                 Paris                    2
3         Mountain View                    2
4                London                    2
5                Berlin                    2
6           Yellowknife                    1
7              Winnipeg                    1
8                Warsaw                    1
9                Vienne                    1
10            Vancouver                    1
11               Tucson                    1
12              Toronto                    1
13  São José dos Campos                    1
14            Stuttgart                    1
15            Stockholm                    1
16               Sidney                    1
17             Santiago                    1
18       Salt Lake City                    1
19                 Rome                    1
20       Rio de Janeiro                    1
21                 Reno                    1
22              Redmond                    1
23                Porto                    1
24               Ottawa                    1
25                 Oslo                    1
26              Orlando                    1
27             New York                    1
28             Montréal                    1
29               Madrid                    1
30              Madison                    1
31                 Lyon                    1
32               Lisbon                    1
33             Helsinki                    1
34              Halifax                    1
35            Frankfurt                    1
36           Fort Worth                    1
37             Edmonton                    1
38           Edinburgh                     1
39               Dublin                    1
40                Dijon                    1
41                Delhi                    1
42            Cupertino                    1
43           Copenhagen                    1
44              Chicago                    1
45         Buenos Aires                    1
46             Budapest                    1
47             Brussels                    1
48             Brasília                    1
49               Boston                    1
50             Bordeaux                    1
51            Bangalore                    1
52            Amsterdam                    1

5f) The cities with the largest number of customers are São Paulo, Prague, Paris, Mountain View, London, and Berlin, with 2 customers in each of those countries.	 

In [48]:
genre_query = \
"""
SELECT tracks.GenreId, genres.Name, COUNT(*) AS "number of tracks"
FROM tracks
JOIN genres ON tracks.GenreId = genres.GenreID
GROUP BY tracks.GenreId
ORDER BY "number of tracks" DESC

"""

with sqlite3.connect('data/chinook.db') as chin_con: 
   genre_df = pd.read_sql(genre_query, chin_con)

genre_df

GenreId                Name  number of tracks
0         1                Rock              1297
1         7               Latin               579
2         3               Metal               374
3         4  Alternative & Punk               332
4         2                Jazz               130
5        19            TV Shows                93
6         6               Blues                81
7        24           Classical                74
8        21               Drama                64
9        14            R&B/Soul                61
10        8              Reggae                58
11        9                 Pop                48
12       10          Soundtrack                43
13       23         Alternative                40
14       17         Hip Hop/Rap                35
15       15   Electronica/Dance                30
16       13         Heavy Metal                28
17       16               World                28
18       20    Sci Fi & Fantasy                26
19       12      Easy Listening                24
20       22              Comedy                17
21       11          Bossa Nova                15
22       18     Science Fiction                13
23        5       Rock And Roll                12
24       25               Opera                 1

5g) The most common genre of tracks is Rock. 